euphonic_wrapper.py

```diff
@@ -0,0 +1,194 @@
+import numpy as np
+import os
+from euphonic import Crystal, ForceConstants, QpointPhononModes, DebyeWaller, ureg
+from euphonic.util import mp_grid
+
+class EuphonicWrapper(object):
+    # This a wrapper around the Euphonic ForceConstants and QpointPhononModes classes to make it easier to access from Matlab
+    # It is meant to be used with a Matlab python_wrapper class and implements a horace_sqw function for use with Horace
+
+    defaults = {'force_constants': None, 'phonon_modes': None, 'debye_waller': None, 'debye_waller_grid': None, 
+                'temperature': 0.0 * ureg('K'), 'bose': True, 'negative_e': False, 'conversion_mat': None, 'chunk': 5000, 
+                'lim': np.inf, 'scattering_lengths': 'Sears1992', 'weights': None, 'asr': None, 'dipole': True, 
+                'eta_scale': 1.0, 'splitting': True, 'insert_gamma': False, 'reduce_qpts': True, 'use_c': False, 
+                'n_threads': 1, 'fall_back_on_python': True, 'convolution_function': 'gauss'}
+
+    def __init__(self, value=None, **kwargs):
+        for key, val in self.defaults.items():
+            setattr(self, key, kwargs.pop(key, self.defaults[key]))
+
+        if value is not None:
+            if hasattr(value, 'calculate_qpoint_phonon_modes'):
+                self.force_constants = value
+            elif hasattr(value, 'calculate_structure_factor'):
+                self.phonon_modes = value
+            elif isinstance(value, dict):
+                kwargs.update(value)
+
+    def calculate_sf(self, qpts):
+        if self.temperature > 0:
+            if self.debye_waller is None and self.debye_waller_grid is not None:
+                self.calculate_debye_waller()
+        phonons = self.calculate_phonon_modes(qpts)
+        sf_obj = phonons.calculate_structure_factor(scattering_lengths=self.scattering_lengths,
+                                                    dw=self.debye_waller)
+        w = sf_obj.frequencies.magnitude
+        sf = sf_obj.structure_factors.magnitude
+        if self.temperature > 0 and self.bose:
+            bose = sf_obj._bose_factor(self.temperature)
+            sf = (1 + bose) * sf
+        if self.negative_e:
+            w = np.hstack((w, -w))
+            neg_sf = sf_obj.structure_factors.magnitude
+            if self.temperature > 0 and self.bose:
+                neg_sf = bose * neg_sf
+            sf = np.hstack((sf, neg_sf))
+        return w, sf
+        
+    def horace_sqw(self, qh, qk, ql, scale=1., *args, **kwargs):
```

Comment:
Is there a reason scale is here as an argument rather than on the class? I think its fine but was just wondering if there's something I hadn't thought of

Reply:
Yeah, it's meant so that you can use it with `multifit` to fit the scale factor - `multifit` has a stupid syntax where it expects the first vector after the coordinates to be a vector of parameters. In this case we would only have one fittable parameter - `scale`... I haven't actually tried to use this wrapper with fitting so this might have to change...